server/wavelength/modules/economy/index.js

```diff
@@ -0,0 +1,124 @@
+// Used with nef
+
+'use strict';
+
+const FS = require(('../../../../.lib-dist/fs')).FS;
+const DB_PATH = 'config/database'; // path to your database folder
+
+module.exports = class Economy {
+	constructor(name, dbName) {
+		this.name = name;
+		this.dbName = dbName;
+		this.db = Db[dbName];
+
+		this.transactionLogs = FS('logs/transactions.txt');
+		this.tLogs = this.transactionLogs.createAppendStream();
+	}
+
+	grammarizeName(value) {
+		return value !== 1 ? `${this.name}s` : this.name;
+	}
+
+	// positive integer
+	isPInteger(value) {
+		let numValue = Number(value);
+
+		return Number.isInteger(numValue) && numValue > 0;
+	}
+
+	get(target) {
+		let tarId = toId(target);
+
+		return this.db.get(tarId, 0);
+	}
+
+	award(target, value) {
+		if (!this.isPInteger(value)) return false;
+
+		let tarId = toId(target);
+
+		this.db.put(tarId, amount => amount + Number(value), 0);
+
+		let amount = this.get(tarId);
+
+		return `${amount} ${this.grammarizeName(amount)}`;
+	}
+
+	remove(target, value) {
+		if (!this.isPInteger(value)) return false;
+
+		let tarId = toId(target);
+
+		this.db.put(tarId, amount => amount - Number(value), 0);
+
+		let amount = this.get(tarId);
+		if (amount <= 0) {
+			this.db.remove(tarId);
+			return `0 ${this.name}s`;
+		}
+
+		return `${amount} ${this.grammarizeName(amount)}`;
+	}
+
+	getStats() {
+		let total = 0;
+		let keys = this.db.keys();
+		let len = keys.length;
```

Comment:
No need to declare this as a variable, array.length is often used in the loop itself.